server/utils/nunjucksSetup.ts

```diff
@@ -245,5 +253,36 @@ export function registerNunjucks(conditionService: ConditionService, app?: expre
     }
   })
 
+  njkEnv.addFilter('dateToDisplay', (licence: Licence) => {
+    const licenceType = licence.typeCode
+    const led = licence.licenceExpiryDate ? toDate(licence.licenceExpiryDate) : null
+    const tussd = licence.topupSupervisionStartDate ? toDate(licence.topupSupervisionStartDate) : null
+    const tused = licence.topupSupervisionExpiryDate ? toDate(licence.topupSupervisionExpiryDate) : null
+
+    let dateToDisplay: Date
+    let textToDisplay = ''
+
+    if (licenceType === 'AP') {
+      textToDisplay = 'Licence end date'
+      dateToDisplay = led
+    }
+
+    if (licenceType === 'AP_PSS') {
+      textToDisplay = isToday(tussd) ? 'PSS end date' : 'Licence end date'
```

Comment:
Does this function accept null values? Its possible tussd could be null here

Reply:
fair point. Have split functionality into 2 if blocks so can can check tussd exists. 
It does bring into question though of what should be rendered if there is a tused but no tussd !